src/app/app.component.ts

```diff
@@ -34,29 +34,29 @@ export class AppComponent implements OnInit, OnDestroy {
 
   changeCharactersInput(element: any): void {
     // 1.1. Add functionality to changeCharactersInput method. Changes searchTermByCharacters Subject value on input change.
-    // const inputValue: string = element.target.value;
-    // this.searchTermByCharacters.next(inputValue);
+    const inputValue: string = element.target.value;
+    this.searchTermByCharacters.next(inputValue);
   }
 
   initCharacterEvents(): void {
     // 1.2. Add API call on each user input. Use mockDataService.getCharacters - to make get request.
     // 2. Since we don't want to spam our service add filter by input value and do not call API until a user enters at least 3 chars.
     // 3. Add debounce to prevent API calls until user stop typing.
-    // this.charactersResults$ = this.searchTermByCharacters.pipe(
-    //   filter((elem: string) => elem.length >= 3),
-    //   debounceTime(500),
-    //   switchMap((elem: string) => this.mockDataService.getCharacters(elem))
-    // );
+    this.charactersResults$ = this.searchTermByCharacters.pipe(
```

Comment:
I think we don't need to recreate this stream each time we hit keyboard, we need update subject content only.

this.searchTermByCharacters.pipe(
       filter((elem: string) => elem.length >= 3),
       debounceTime(500),
       switchMap((elem: string) => this.mockDataService.getCharacters(elem))); might be stored as class property since tempalte already subscribe to it via async pipe